components/recepiesSelector/RecepiesSelector.tsx

```diff
@@ -0,0 +1,41 @@
+'use client';
+
+import Image from 'next/image';
+import React, { useState } from 'react';
+
+import getCategoriesList from './helpers/getCategoriesList';
+import filterIcon from '../../public/assets/icons/Filters.svg';
+import sortIcon from '../../public/assets/icons/Sort.svg';
+
+export default function RecepiesSelector() {
+  const [activeCat, setActiveCat] = useState('all');
+  const [recipesNumb, setRecipesNumb] = useState(8);
+
+  const catList = getCategoriesList({ activeCat, setActiveCat });
+
```

Comment:
I dont like this pattern.
I think it would be better to make something like this with separate ```<ListItem />``` component

```tsx
categories.map(cat => <ListItem Elem={cat.Elem} title={cat.title}, key={cat.key}/>)
```

```tsx
function ListItem({ Elem, title, key }) {
    return <li>
        <button
          className={`${
            isActive ? 'text-accent-600' : 'text-primary-900'
          } hover:text-accent-400 transition-all duration-200 myRecepBtn flex flex-col justify-center items-center whitespace-nowrap gap-4`}
          onClick={() => setActiveCat(key)}
        >
          <Elem color={isActive ? 'fill-accent-600' : 'fill-primary-900'} />
          {title}
        </button>
      </li>
}
```